BoxContentSDK/BoxContentSDK/Requests/BOXFileRepresentationDownloadRequest.m

```diff
@@ -0,0 +1,114 @@
+//
+//  BOXRepresentationDownloadRequest.m
+//  BoxContentSDK
+//
+//  Created by Clement Rousselle on 3/2/17.
+//  Copyright © 2017 Box. All rights reserved.
+//
+
+#import "BOXFileRepresentationDownloadRequest.h"
+#import "BOXRepresentation.h"
+#import "BOXAPIDataOperation.h"
+#import "BOXRequest_Private.h"
+
+@interface BOXFileRepresentationDownloadRequest ()
+
+@property (nonatomic, readonly, strong) NSString *destinationPath;
+@property (nonatomic, readonly, strong) NSOutputStream *outputStream;
+@property (nonatomic, readonly, strong) NSString *fileID;
+@property (nonatomic, readwrite, strong) BOXRepresentation *representation;
+@end
+
+@implementation BOXFileRepresentationDownloadRequest
+
+- (instancetype)initWithLocalDestination:(NSString *)destinationPath
+                                  fileID:(NSString *)fileID
+                          representation:(BOXRepresentation *)representation
+{
+    if (self = [super init]) {
+        _destinationPath = destinationPath;
+        _fileID = fileID;
+        _representation = representation;
+    }
+    return self;
+}
+
+- (instancetype)initWithOutputStream:(NSOutputStream *)outputStream
+                              fileID:(NSString *)fileID
+                      representation:(BOXRepresentation *)representation
+{
+    if (self = [super init]) {
+        _outputStream = outputStream;
+        _fileID = fileID;
+        _representation = representation;
+    }
+    return self;
+}
+
+- (BOXAPIOperation *)createOperation
+{
+    NSURL *URL = self.representation.contentURL;
+    
+    
+    BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                         HTTPMethod:BOXAPIHTTPMethodGET
+                                              queryStringParameters:nil
+                                                     bodyDictionary:nil
+                                                       successBlock:nil
+                                                       failureBlock:nil];
+    
+    BOXAssert(self.outputStream != nil || self.destinationPath != nil, @"An output stream or destination file path must be specified.");
+    BOXAssert(!(self.outputStream != nil && self.destinationPath != nil), @"You cannot specify both an outputStream and a destination file path.");
+    
+    if (self.outputStream != nil) {
+        dataOperation.outputStream = self.outputStream;
+    } else {
+        dataOperation.outputStream = [[NSOutputStream alloc] initToFileAtPath:self.destinationPath append:NO];
```

Comment:
I think we need to support background downloads, yes. It should be handled similarly to our current download operation, I'm just unsure of what needs to be done for it.
Is it something we have to modify now, or that the URL session team will handle later on ? 
@thuyBox